django_tasks/backends/database/management/commands/db_worker.py

```diff
@@ -158,11 +159,8 @@ def valid_backend_name(val: str) -> str:
 
 
 def valid_interval(val: str) -> float:
-    # Cast to an int first to catch invalid values like 'inf'
-    int(val)
-
     num = float(val)
-    if num < 0:
+    if not math.isfinite(num) or num < 0:
         raise ArgumentTypeError("Must be greater than zero")
```

Comment:
Nit: _technically_ `+inf` is greater than zero :wink: Perhaps it should be 2 checks with different messages?